### PR TITLE
EZP-27568: Define the markup for editing supported field types

### PR DESCRIPTION
**Pull request by dpobel**
JIRA: https://jira.ez.no/browse/EZP-27578

# Description

Specifies how supported Field Types should be rendered when edited.

* [x] Initialize document with current status
* [x] Authors `ezauthor`
* [x] File `ezbinaryfile`
* [x] Checkbox `ezboolean`
* [x] Country `ezcountry`
* [x] Date `ezdate`
* [x] Date and Time `ezdatetime`
* [x] Email Address `ezemail`
* [x] Float `ezfloat`
* [x] Map Location `ezgmaplocation`
* [x] Image `ezimage`
* [x] Integer `ezinteger`
* [x] ISBN `ezisbn`
* [x] Keywords `ezkeyword`
* [x] Media `ezmedia`
* [x] Content Relation `ezobjectrelation` (done but to be detailed/deepened later)
* [x] Content Relations `ezobjectrelationlist` (done but to be detailed/deepened later)
* [ ] ~~Page `ezpage`~~
* [x] Rich Text `ezrichtext`
* [x] Selection `ezselection`
* [x] Text Line `ezstring` 
* [x] Text Block `eztextblock` 
* [x] Time `eztime`
* [x] URL `ezurl`
* [x] User Accout `ezuser`
* [x] Server side validation

### CODE REVIEW

File: docs/specs/field_edit_markup.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+If the Field has a description, this description is added in a paragraph as the
+last element of the `div.ez-field-edit`. If the Field is not translatable and
+the user is translating a Content, the description is replaced with the non
+translatable notice.
+
+So for a required Field with a description, the base structure is:
+
+```html
+<div class="ez-field-edit ez-field-edit-ezstring ez-field-edit-required">
+    <p class="ez-field-definition-name"><label for="autogenerateid">Field Defintion Name</label></p>
```
Comment (dpobel) [inline docs/specs/field_edit_markup.md]: self note: this is also where validation messages appear. Open question: how to share message between server and frontend?

File: docs/specs/field_edit_markup.md [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+If the Field has a description, this description is added in a paragraph as the
+last element of the `div.ez-field-edit`. If the Field is not translatable and
+the user is translating a Content, the description is replaced with the non
+translatable notice.
+
+So for a required Field with a description, the base structure is:
+
+```html
+<div class="ez-field-edit ez-field-edit-ezstring ez-field-edit-required">
+    <div class="ez-field-edit-text-zone">
```
Comment (dpobel) [inline docs/specs/field_edit_markup.md:60]: self note: this is also where validation messages appear. Open question: how to share message between server and frontend?

File: docs/specs/field_edit_markup.md [hunk 3]
```diff
@@ -0,0 +1,1295 @@
...
+        </label>
+    </div>
+    <div class="ez-field-edit-ui">
+        <!--
+            specific to Field Type part where input, select, textarea are marked
+            as read only with `readonly` attribute and buttons are disabled with
+            the `disabled` attribute.
+        -->
+    </div>
+    <p class="ez-field-not-translatable">This is not a translatable field and cannot be modified</p>
```
Comment (SylvainGuittard) [inline docs/specs/field_edit_markup.md]: We simplified the sentence for non translatable fields.
Can we do something like this?
`Translation for %name of the field% is disabled`

File: docs/specs/field_edit_markup.md [hunk 4]
```diff
@@ -0,0 +1,1295 @@
...
+As for others Field Type, the outermost `div` gets the `ez-field-edit-required`
+class and the `input` gets the `required` attribute.
+
+
+## Date and Time `ezdatetime`
+
+### Options
+
+* Required
+* Use seconds
```
Comment (SylvainGuittard) [inline docs/specs/field_edit_markup.md:366]: Thanks, I learnt something: `Date and Time` FieldType can have seconds.

File: docs/specs/field_edit_markup.md [hunk 5]
```diff
@@ -0,0 +1,1295 @@
...
+
+```html
+<div class="ez-field-edit ez-field-edit-ezbinaryfile">
+    <div class="ez-field-edit-text-zone">
+        <label class="ez-field-definition-name" for="auto-generated-id">
+            Field Definition Name
+        </label>
+    </div>
+    <div class="ez-field-edit-ui">
+        <input type="file" accept="image/jpg, image/gif, image/png" id="auto-generated-id" name="auto-generated-name">
```
Comment (SylvainGuittard) [inline docs/specs/field_edit_markup.md:619]: I am bringing back the SVG topic. Can we add `svg` to `accept`?

File: docs/specs/field_edit_markup.md [hunk 6]
```diff
@@ -0,0 +1,1295 @@
...
+                </div>
+            </div>
+            <div class="ez-sub-field ez-sub-field-longitude">
+                <div class="ez-sub-field-text-zone">
+                    <label class="ez-sub-field-name" for="auto-generated-id-lon">
+                        Longitude
+                    </label>
+                </div>
+                <div class="ez-sub-field-ui">
+                    <input type="number" step="any" min="-90" max="90" id="auto-generated-id-lon" name="auto-generated-name-lon">
```
Comment (SylvainGuittard) [inline docs/specs/field_edit_markup.md:572]: Will the user be able to set the value directly?

**Comment by SylvainGuittard** [comment]
I really like the structure of the code. It's really clear compared to the current markup (for some Field Types).
I also really like the approach you took for `ez-sub-field-ui`.

File: docs/specs/field_edit_markup.md (hunk 3 above)
Comment (dpobel) [reply docs/specs/field_edit_markup.md]: of course we can :)

File: docs/specs/field_edit_markup.md (hunk 6 above)
Comment (dpobel) [reply docs/specs/field_edit_markup.md:572]: with the server side generated markup only yes, then when we'll apply frontend enhancement with the ability the pick a Location in a map and while doing that we *can* restrict manual input (even if I don't really see the point in putting this restriction in place)

File: docs/specs/field_edit_markup.md (hunk 5 above)
Comment (dpobel) [reply docs/specs/field_edit_markup.md:619]: The problem is not only in adding svg mimetype here. The problem that the Image Field Type is not able to handle SVG document.
See also discussion in v1 bug https://jira.ez.no/browse/EZP-25208 and related story: https://jira.ez.no/browse/EZP-25422 (I'm sure there are others)

**Comment by webhdx** [comment]
@dpobel How about `Page` FieldType? Is it going to be deprecated?

**Comment by dpobel** [comment]
@webhdx indeed I forgot to explain why it's not in that document. It's not deprecated (very unlikely, it's one of the base for *studio*/*platform EE*) :)) but the expected UI behavior to edit Page fields without EE is not specified at all, so it's impossible to guess which kind of markup is needed for it.

File: docs/specs/field_edit_markup.md [hunk 7]
```diff
@@ -0,0 +1,1295 @@
...
+
+For a Content Type with a Text Line, a Map Location and a Text Block fields, the
+Content edit form would look like:
+
+```html
+<form>
+  <ez-field-edit fieldtype="ezstring">
+     <!-- markup for Text Line -->
+  </ez-field-edit>
+  <ez-field-edit-maplocation fieldtype="ezgmaplocation">
```
Comment (andrerom) [inline docs/specs/field_edit_markup.md:20]: ez-field-edit?

**Pull request merged**